dronegen/container_images.go

```diff
@@ -0,0 +1,942 @@
+		Flags: &TriggerFlags{
+			ShouldAffectProductionImages: false,
+			ShouldBuildNewImages:         true,
+			UseUniqueStagingTag:          false,
+		},
+		SupportedVersions: []*releaseVersion{
+			{
+				MajorVersion:        branchMajorVersion,
+				ShellVersion:        "$DRONE_TAG",
+				RelativeVersionName: "branch",
+			},
+		},
+	}
+}
+
+func NewPromoteTrigger(branchMajorVersion string) *TriggerInfo {
+	promoteTrigger := triggerPromote
+	promoteTrigger.Target.Include = append(promoteTrigger.Target.Include, "promote-docker")
+
+	return &TriggerInfo{
+		Trigger: promoteTrigger,
+		Name:    "promote",
+		Flags: &TriggerFlags{
+			ShouldAffectProductionImages: true,
+			ShouldBuildNewImages:         false,
+			UseUniqueStagingTag:          false,
+		},
+		SupportedVersions: []*releaseVersion{
+			{
+				MajorVersion:        branchMajorVersion,
+				ShellVersion:        "$DRONE_TAG",
+				RelativeVersionName: "branch",
+			},
+		},
+		SetupSteps: verifyValidPromoteRunSteps(),
+	}
+}
+
+func NewCronTrigger(latestMajorVersions []string) *TriggerInfo {
+	if len(latestMajorVersions) == 0 {
+		return nil
+	}
+
+	majorVersionVarDirectory := "/go/vars/full-version"
+
+	supportedVersions := make([]*releaseVersion, 0, len(latestMajorVersions))
+	if len(latestMajorVersions) > 0 {
+		latestMajorVersion := latestMajorVersions[0]
+		supportedVersions = append(supportedVersions, &releaseVersion{
+			MajorVersion:        latestMajorVersion,
+			ShellVersion:        readCronShellVersionCommand(majorVersionVarDirectory, latestMajorVersion),
+			RelativeVersionName: "current-version",
+			SetupSteps:          []step{getLatestSemverStep(latestMajorVersion, majorVersionVarDirectory)},
+		})
+
+		if len(latestMajorVersions) > 1 {
+			for i, majorVersion := range latestMajorVersions[1:] {
+				supportedVersions = append(supportedVersions, &releaseVersion{
+					MajorVersion:        majorVersion,
+					ShellVersion:        readCronShellVersionCommand(majorVersionVarDirectory, majorVersion),
+					RelativeVersionName: fmt.Sprintf("previous-version-%d", i+1),
+					SetupSteps:          []step{getLatestSemverStep(majorVersion, majorVersionVarDirectory)},
+				})
+			}
+		}
+	}
+
+	return &TriggerInfo{
+		Trigger: cronTrigger([]string{"teleport-container-images-cron"}),
+		Name:    "cron",
+		Flags: &TriggerFlags{
+			ShouldAffectProductionImages: true,
+			ShouldBuildNewImages:         true,
+			UseUniqueStagingTag:          true,
+		},
+		SupportedVersions: supportedVersions,
+	}
+}
+
+func getLatestSemverStep(majorVersion string, majorVersionVarDirectory string) step {
+	// We don't use "/go/src/github.com/gravitational/teleport" here as a later stage
+	// may need to clone a different version, and "/go" persists between steps
+	cloneDirectory := "/tmp/teleport"
+	majorVersionVarPath := path.Join(majorVersionVarDirectory, majorVersion)
+	return step{
+		Name:  fmt.Sprintf("Find the latest available semver for %s", majorVersion),
+		Image: "golang:1.18",
```

Comment:
We are now using Go 1.19 in master and v11. Is there a constant defined somewhere with Go version we can use instead of hardcoding it inline here?